### ISSUE TITLE
Details about tf.keras.layers.Conv2DTranspose params 'padding' and 'output_padding'

### ISSUE THREAD

**Issue by beginner401** [tensorflow/tensorflow#53107]
https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2DTranspose

## Description of issue (what needs changing):
The documentation tells developers that output_padding has only one restriction that it should be lower that stride param, and it may not be all-inclusive.

### Clear description
Different from PyTorch, Conv2DTranspose implements Keras API, which uses padding params as 'same' or 'valid' instead of concrete int values. That gives developers a false impression that we can give padding cal tasks safely to TF. However, padding param should work with output_padding param, and output_padding cannot be set casually. I check the source code of TF and found that while using padding params as 'same', TF set pad to be kernel_size / 2 defaultly, that means output_padding param shoud be set to stride - 1, or mistakes may occur in some situation. However, the documentation tells developers that output_padding has only one restriction that it should be lower that stride param, so it may not be all-inclusive.

### Suggestions
Actually, setting param 'padding' in 'same' and 'valid' in this layer may be an ill considered decision, but since TF should implement Keras API, may be we have two solutions: One is to tell the developers in the documentation the relation of output_padding with padding param, and adds the implementation theory of same padding in this layer; Or just to adapt the actual pad values to the output_padding param as far as possible, and to warn the developers the risk and give them suggestions.

**Comment by mohantym** [comment]
Hi @beginner401!
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by beginner401** [comment]
Hi @mohantym !
I have post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues), and thanks for your guidance.
